tests/unittests/RecommendationSystemTest.cpp

```diff
@@ -715,16 +721,69 @@ class RecommendationSystemTest : public BackendTest {
     }
   }
 
+  void dispatchInference(HostManager *hostManager, ExecutionContext &context) {
+    // If additional requests are desired, setup additional contexts.
+    std::vector<std::unique_ptr<ExecutionContext>> contexts;
+    std::unique_ptr<ExecutionContext> originalContextPtr(&context);
+    contexts.push_back(std::move(originalContextPtr));
+    if (concurrentReqestsOpt > 1) {
+      // Clone the placeholder bindings into a new executionContext.
+      for (unsigned i = 0, max = concurrentReqestsOpt - 1; i < max; i++) {
+        std::unique_ptr<ExecutionContext> newContext =
+            llvm::make_unique<ExecutionContext>(
+                llvm::make_unique<PlaceholderBindings>(
+                    context.getPlaceholderBindings()->clone()));
+        contexts.push_back(std::move(newContext));
+      }
+    }
+    std::vector<std::promise<void>> promises(concurrentReqestsOpt);
+    std::vector<std::future<void>> futures;
+    for (auto &promise : promises) {
+      futures.push_back(promise.get_future());
+    }
+    for (unsigned i = 0; i < concurrentReqestsOpt; i++) {
+      hostManager->runNetwork(
+          "main", std::move(contexts[i]),
+          [&contexts, &promises,
+           i](runtime::RunIdentifierTy, Error err,
+              std::unique_ptr<ExecutionContext> contextPtr) {
+            contexts[i] = std::move(contextPtr);
+            // Expect no errors.
+            EXIT_ON_ERR(std::move(err));
+            promises[i].set_value();
+          });
+    }
+
+    for (auto &future : futures) {
+      future.wait();
+    }
+    // Release the original context passed in by reference so we don't free it.
+    contexts[0].release();
+  }
+
+  std::vector<std::unique_ptr<runtime::DeviceConfig>>
```

Comment:
Please add a Doxygen comment. BTW, should this function (and may be a couple of other functions) be static or reside in an anonymous namespace?